pkg/componentreadiness/tracker/tracker.go

```diff
@@ -0,0 +1,280 @@
+	"cloud.google.com/go/bigquery"
+	"github.com/google/uuid"
+	"github.com/openshift/sippy/pkg/apis/api"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/api/iterator"
+)
+
+const (
+	testRegressionsDataSet = "ci_analysis_us"
+	testRegressionsTable   = "test_regressions"
+)
+
+type Storage interface {
+	ListCurrentRegressions(release string) ([]api.TestRegression, error)
+	OpenRegression(release string, newRegressedTest api.ComponentReportTestSummary) (*api.TestRegression, error)
+	ReOpenRegression(regressionID string) error
+	CloseRegression(regressionID string, closedAt time.Time) error
+}
+
+// BigQueryStorage is the primary implementation for real world usage, tracking when we detect test
+// regressions opening/closing in the Bigquery test_regressions table.
+type BigQueryStorage struct {
+	client *bigquery.Client
+}
+
+func NewBigQueryStorage(client *bigquery.Client) Storage {
+	return &BigQueryStorage{client: client}
+}
+
+func (bqs *BigQueryStorage) ListCurrentRegressions(release string) ([]api.TestRegression, error) {
+	// List open regressions (no closed date), or those that closed within the last two days. This is to prevent flapping
+	// and return more accurate opened dates when a test is falling in / out of the report.
+	queryString := fmt.Sprintf("SELECT * FROM %s.%s WHERE release = @SampleRelease AND (closed IS NULL or closed > TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 2 DAY))",
+		testRegressionsDataSet, testRegressionsTable)
+
+	params := make([]bigquery.QueryParameter, 0)
+	params = append(params, []bigquery.QueryParameter{
+		{
+			Name:  "SampleRelease",
+			Value: release,
+		},
+	}...)
+
+	sampleQuery := bqs.client.Query(queryString)
+	sampleQuery.Parameters = append(sampleQuery.Parameters, params...)
+
+	regressions := make([]api.TestRegression, 0)
+	log.Infof("Fetching current test regressions with:\n%s\nParameters:\n%+v\n",
+		sampleQuery.Q, sampleQuery.Parameters)
+
+	it, err := sampleQuery.Read(context.TODO())
+	if err != nil {
+		log.WithError(err).Error("error querying triaged incidents from bigquery")
+		return regressions, err
+	}
+
+	for {
+		var regression api.TestRegression
+		err := it.Next(&regression)
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			log.WithError(err).Error("error parsing triaged incident from bigquery")
+			return nil, errors.Wrap(err, "error parsing triaged incident from bigquery")
+		}
+		regressions = append(regressions, regression)
+	}
+	return regressions, nil
+
+}
+func (bqs *BigQueryStorage) OpenRegression(release string, newRegressedTest api.ComponentReportTestSummary) (*api.TestRegression, error) {
+	id := uuid.New()
+	newRegression := &api.TestRegression{
+		Release:      release,
+		TestID:       newRegressedTest.TestID,
+		TestName:     bigquery.NullString{StringVal: newRegressedTest.TestName, Valid: true},
+		RegressionID: id.String(),
+		Opened:       time.Now(),
+		Variants: []api.TriagedVariant{
+			{
+				Key:   "Network",
```

Comment:
Feel free to take it or leave it, I created some [variant constants](https://github.com/openshift/sippy/blob/f41dff9ef1f23735c8ac2e1b5939caa1bcc0baeb/pkg/componentreadiness/resolvedissues/types.go#L13) with a comment to sync up with your [variantregistry constants](https://github.com/openshift/sippy/blob/1025904eb603f8cab6173c498e491ecc7bd5272f/pkg/variantregistry/ocp.go#L241).  Looks like yours merged, should we use them here and I can update mine in another PR?